### PR TITLE
add bakta module

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes https://github.com/nf-core/modules/issues/1079 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

Bakta requires a 40+ GB input database so I had to use `-stub-run` for testing. I plan on using this extensively so I will relay any issues that might pop up that the `stubs` do not catch.

### CODE REVIEW

File: tests/modules/bakta/test.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+## TODO nf-core: Please run the following command to build this file:
+#                nf-core modules create-test-yml bakta
+- name: bakta
+  command: nextflow run ./tests/modules/bakta -entry test_bakta -c tests/config/nextflow.config -stub-run
+  tags:
+    - bakta
+  files:
+    - path: output/bakta/test.embl
+      md5sum: d41d8cd98f00b204e9800998ecf8427e
```
Comment (FriederikeHanssen) [inline tests/modules/bakta/test.yml:9]: all the test output files are emtpy. Can the tool not be run with the avaible test data?

Comment (rpetit3) [reply tests/modules/bakta/test.yml:9]: I realize this is not optimal, but Bakta requires a 40GB+ input database, so I had to go with `-stub-run`. 

If its any consolation, I've implemented a test on my end that uses this module. https://github.com/bactopia/bactopia/blob/dsl2/subworkflows/local/bakta/test.yml (this done using a local runner for GitHub Actions, so its actually testing with the 40GB database)

Comment (FriederikeHanssen) [reply tests/modules/bakta/test.yml:9]: I see. OK then I'll approve the PR

**Review by FriederikeHanssen — APPROVED** [review]

**Comment by oschwengers** [comment]
@rpetit3 Thanks a lot for building and adding this - highly appreciated!

FYI: I just added a lacking AMRFinderPlus mock database: https://github.com/oschwengers/bakta/commit/a3c514dca661dd4bbd60aa5635a1a72e1dec393d

Maybe the module could now use the test mock database available in the Bakta repository in combination with any small test sequence (we have test sequences in the Bakta repo, as well).

**Pull request merged**